tailcfg/tailcfg.go

```diff
@@ -492,15 +493,31 @@ var FilterAllowAll = []FilterRule{
 	},
 }
 
+// DNSConfig is the DNS configuration.
+type DNSConfig struct {
+	Nameservers []netaddr.IP `json:",omitempty"`
+	Domains     []string     `json:",omitempty"`
+	PerDomain   bool
+	Proxied     bool
+}
+
 type MapResponse struct {
 	KeepAlive bool // if set, all other fields are ignored
 
 	// Networking
-	Node        *Node
-	Peers       []*Node
-	DNS         []wgcfg.IP
+	Node    *Node
+	Peers   []*Node
+	DERPMap *DERPMap
+
+	// DNS is the same as DNSConfig.Nameservers.
+	// TODO(dmytro): when all devices are running a recent enough version,
```

Comment:
Instead of the TODO, mark this as Deprecated. https://github.com/golang/go/wiki/Deprecated